src/main/java/com/synopsys/integration/detect/workflow/componentlocationanalysis/GenerateComponentLocationAnalysisOperation.java

```diff
@@ -57,11 +64,23 @@ public void locateComponentsForOnlineIntelligentScan() throws ComponentLocatorEx
         logger.info("Intelligent Scan mode does not support Component Location Analysis.");
         throw new ComponentLocatorException("Failed to generate Component Location Analysis file.");
     }
+    
+    private File serializeInputToJson(File saveInputFileDir, Input libInput) throws DetectUserFriendlyException {
+        Gson gson = new GsonBuilder().setPrettyPrinting().create();
+        String serializedLibInput = gson.toJson(libInput);
+        try {
+            File componentsSourceInputFile =  new File (saveInputFileDir, "components-source.json");
+            DetectFileUtils.writeToFile(componentsSourceInputFile, serializedLibInput);
+            return componentsSourceInputFile;
+        } catch (IOException ex) {
+            throw new DetectUserFriendlyException("Failed to create component location analysis output file", ex, ExitCodeType.FAILURE_UNKNOWN_ERROR);
```

Comment:
This will fail the Detect build, when the creation of this file (or any failure during component location analysis) should never fail Detect. Will approve since this will be followed immediately by another PR for adding the debug level restriction and **_we will not release a QA build before that subsequent PR_**.